### DISCUSSION TITLE
IKE SA keep initializing when setting left/right subnet to 0.0.0.0/0 on both ends

### DISCUSSION THREAD

**Discussion by Ruoyu-y** [strongswan/strongswan#571] [Q&A]
I have two endpoints each working as the gateway for a local cluster. So i would like to have all traffic go thru the ipsec tunnel. To fulfill that, i set both left/right subnet to 0.0.0.0/0. But after setting that, i could see the ipsec tunnel established and get deleted in about 15 seconds. And this kind of situation keeps rotating.
Here attaches my configs.
Left side:
conn hubaoverlayhub-Connhubaoverlayhub
  left=%any
  right=10.10.70.49
  ikelifetime=3h
  lifetime=1h
  margintime=9m
  keyingtries=%forever
  dpdaction=restart
  dpddelay=30s
  closeaction=restart
  leftauth=pubkey
  rightauth=pubkey
  leftcert=/etc/ipsec.d/certs/hubaoverlayhub_public.pem
  leftsendcert=yes
  rightsendcert=yes
  rightsubnet=0.0.0.0/0
  leftsubnet=0.0.0.0/0
  auto=start
  leftid="CN=hub-huba-cert"
  rightid="CN=hub-overlayhub-cert"
  leftupdown=/etc/updown
  keyexchange=ikev2
  mark=30
  esp=aes128-sha256-modp3072,aes256-sha256-modp3072
  ike=aes128-sha256-modp3072,aes256-sha256-modp3072
  type=tunnel

Right side:
conn hubaoverlayhub-Connhubaoverlayhub
  left=%any
  right=10.10.70.39
  ikelifetime=3h
  lifetime=1h
  margintime=9m
  keyingtries=%forever
  dpdaction=restart
  dpddelay=30s
  closeaction=restart
  leftauth=pubkey
  rightauth=pubkey
  leftcert=/etc/ipsec.d/certs/hubaoverlayhub_public.pem
  leftsendcert=yes
  rightsendcert=yes
  rightsubnet=0.0.0.0/0
  leftsubnet=0.0.0.0/0
  auto=start
  rightid="CN=hub-huba-cert"
  leftid="CN=hub-overlayhub-cert"
  leftupdown=/etc/updown
  keyexchange=ikev2
  mark=30
  esp=aes128-sha256-modp3072,aes256-sha256-modp3072
  ike=aes128-sha256-modp3072,aes256-sha256-modp3072
  type=tunnel

I could see these notifications inside the log:
schedule delete of duplicate IKE_SA for peer 'CN=hub-huba-cert' due to uniqueness policy and suspected reauthentication
Then the IKE SA got destroyed.

Any one know what's the problem here? Thanks a lot!

**Comment by tobiasbrunner** [comment]
>  schedule delete of duplicate IKE_SA for peer 'CN=hub-huba-cert' due to uniqueness policy and suspected reauthentication

That happens because you configured both peers with `auto=start` (i.e. they initiate the connection concurrently) and you didn't set `uniqueids=never` on at least one of them (the default is `uniqueids=yes`, which does not allow more than one IKE_SA with the same identities).

**Reply by Ruoyu-y** [reply]
> > schedule delete of duplicate IKE_SA for peer 'CN=hub-huba-cert' due to uniqueness policy and suspected reauthentication
> 
> That happens because you configured both peers with `auto=start` (i.e. they initiate the connection concurrently) and you didn't set `uniqueids=never` on at least one of them (the default is `uniqueids=yes`, which does not allow more than one IKE_SA with the same identities).

Thanks for the answer @tobiasbrunner . But after change one of side with config auto=add and both ends with uniqueids=never. I still see the attempt to perform a second IKEv2 exchange. Any reason for this one? Thanks!

**Reply by tobiasbrunner** [reply]
Read the log.
